npm/create-cypress-tests/src/component-testing/init-component-testing.ts

```diff
@@ -0,0 +1,225 @@
+import fs from 'fs-extra'
+import path from 'path'
+import chalk from 'chalk'
+import inquirer from 'inquirer'
+import highlight from 'cli-highlight'
+import { Template } from './templates/Template'
+import { guessTemplate } from './templates/guessTemplate'
+import { installFrameworkAdapter } from './installFrameworkAdapter'
+import { injectImportSupportCode, injectPluginsCode, getPluginsSourceExample } from './babel/babelTransform'
+import { installDependency } from '../utils'
+
+async function injectOrShowConfigCode (injectFn: () => Promise<boolean>, {
+  code,
+  filePath,
+  fallbackFileMessage,
+  language,
+}: {
+  code: string
+  filePath: string
+  language: string
+  fallbackFileMessage: string
+}) {
+  const fileExists = fs.existsSync(filePath)
+  const readableFilePath = fileExists ? path.relative(process.cwd(), filePath) : fallbackFileMessage
+
+  const printCode = () => {
+    console.log()
+    console.log(highlight(code, { language }))
+    console.log()
+  }
+
+  const printSuccess = () => {
+    console.log(`✅  ${chalk.bold.green(readableFilePath)} was updated with the following config:`)
+    printCode()
+  }
+
+  const printFailure = () => {
+    console.log(`❌  ${chalk.bold.red(readableFilePath)} was not updated automatically. Please add the following config manually: `)
+    printCode()
+  }
+
+  if (!fileExists) {
+    printFailure()
+
+    return
+  }
+
+  // something get completely wrong when using babel or something. Print error message.
+  const injected = await injectFn().catch(() => false)
+
+  injected ? printSuccess() : printFailure()
+}
+
+async function injectAndShowCypressJsonConfig (
+  cypressJsonPath: string,
```

Comment:
This might get a whole lot more complex with this PR: https://github.com/cypress-io/cypress/pull/15280

Since we cannot assume JSON. :thinking:

Reply:
Yeah, that may be a pain in the ass. But I think we can resolve this later, but we already using babel so injecting a new key should be pretty easy